src/eko/runner/legacy.py

```diff
@@ -92,7 +92,7 @@ def __init__(
             couplings=new_theory.couplings,
             order=new_theory.order,
             method=couplings_mod_ev(new_operator.configs.evolution_method),
-            masses=masses,
+            masses=np.array(masses) / new_theory.xif**2,
```

Comment:
This is changing everything, and in particular also the MHOU computed by @andreab1997 

In principle, if it was a bug is a good thing. But why do you believe this should be so general?
I'm not sure why, if I change the factorization scale, the quark masses values should change in the $\alpha_s$ running...

Reply:
Yes this is really strange. Moreover you are rescaling them by `xif` squared but the masses seem to be linear. If this fixes the benchmark, it suggestes a bug somewhere else...

Reply:
I think masses are squared as well

Reply:
Indeed they are: https://github.com/NNPDF/eko/blob/4f9920fd501153812f88d86c1cde20a3694a0e82/src/eko/runner/legacy.py#L75

Reply:
This is also consistent with what we have in tests: https://github.com/NNPDF/eko/blob/4f9920fd501153812f88d86c1cde20a3694a0e82/benchmarks/eko/benchmark_strong_coupling.py#L443

the other place where `Couplings` is called is in `msbar_masses.py` , but since SV with `\bar{MS}` masses are not supported, that should be it.

Reply:
Discussing with @felixhekhorn we convinced ourselves that whatever we do for `alpha_s` in scheme A we should do exactly the same in scheme B. In fact, there is no reason to do different things for the two schemes in that they are doing basically the same evolution (the only difference being that scheme A is using the varied splitting functions and scheme B is not).  Note that (according to @felixhekhorn ) in `eko==0.10` all the masses were shifted in this way, including also scheme B. In particular, in scheme B we have `as1` that is used (correctly) for both the evolution and the "additional sv term" and we actually need  to compute it rescaling the masses (as we were doing before). So, according to us, scheme B in `eko==0.12` is now bugged. Do you agree?

Reply:
Yes I agree on both counts